File: Bridge_Pattern/go/Bridge_Pattern.go
package Bridge_Pattern

import "fmt"

type printer interface {
	printFile()
}
type computer interface {
	print()
	setPrinter(printer)
}

//linux
type linux struct {
	printer printer
}

func (l *linux) print() {
	fmt.Println("Print request for linux")
	l.printer.printFile()
}

func (l *linux) setPrinter(p printer) {
	l.printer = p
}

//windows

type windows struct {
	printer printer
}

func (w *windows) print() {
	fmt.Println("Print request for windows")
	w.printer.printFile()
}
func (w *windows) setPrinter(p printer) {
	w.printer = p
}

//hp
type hp_windows struct {
}

func (p *hp_windows) printFile() {
	fmt.Println("OS:windows, Printing by a HP Printer")
}

type hp_linux struct {
}

func (p *hp_linux) printFile() {
	fmt.Println("OS:linux, Printing by a HP Printer")
}

//ms
type suf_windows struct {
}

func (s *suf_windows) printFile() {
	fmt.Println("OS:windows, Print by a Suf Printer")
}

File: Bridge_Pattern/cpp/Bridge_Pattern.cc
#include<iostream>
using namespace std;
class printer {
public:
    virtual void printFile() = 0;
};

class compter {
public:
    virtual void print() = 0;
    virtual void setPrinter(printer* p) = 0;
};

class windows : public compter {
public:
    void print() {
        cout << "Print request for windows" << endl;
        p->printFile();
    }
    void setPrinter(printer* p) {
        this->p = p;
    }
private:
    printer* p;
};

class linux : public compter {
public:
    void print() {
        cout << "Print request for linux" << endl;
        p->printFile();
    }
    void setPrinter(printer* p) {
        this->p = p;
    }
private:
    printer* p;
};

class hp_windows : public printer {
public:
    void printFile() {
        cout << "OS:windows Printing by a HP Printer" << endl;
    }
};
class hp_linux : public printer {
public:
    void printFile() {
        cout << "OS:linux Printing by a HP Printer" << endl;
    }
};
class suf_windows : public printer{
public:
    void printFile() {
        cout << "OS:windows Printing by a Suf Printer" << endl;
    }
};

void test_bridge() {
    hp_windows hp_Printer;
    suf_windows suf_windows;
    
    windows win;
    linux linus;

    linus.setPrinter(&hp_Printer);
    win.setPrinter(&suf_windows);

    linus.print();
    win.print();
}
int main() {
    test_bridge();
    return 0;
}